### PR TITLE
Library branch

### PR DESCRIPTION
**Pull request by Luzaks**
In this pull request we created a library using:

- HTML/CSS for structure and style.

- JavaScript for functionality using ES6 standards for arrays, constructors, functions.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "parserOptions": {
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "rules": {
+    "arrow-parens": "off",
+    "no-unused-vars": "off",
+    "no-unused-expressions" : "off",
+    "no-nested-ternary" : "off"
+  }
```
Comment (MauricioRobayo) [inline .eslintrc.json]: Please use the provided [.eslintrc.json](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json) and do not modify it.

**Review by MauricioRobayo — CHANGES_REQUESTED** [review]
You are doing a great job, the project is looking very well, and good design skills showing up here!

But... we still need to double-check the following before continuing:

* [ ] Please do not allow to add and empty book, it doesn't make much sense :D.
* [ ] Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))

Keep up the good work, the project is really nice!

Status: __Request changes__.

Optional: You could do better with the commit messages, having [Linter] Linter round 2, [Linter] Linter round 3, .. 4, ... 5, ... 6 does not look very professional, remember that a potential employer could take a look at your work, you could use [`git rebase` with the interactive option](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) to squash some of those commits into one and have a cleaner commit history.

**Comment by MarioDena** [comment]
Hello @MauricioRobayo. I rebased the commit messages.

I added a rule to disallow the creation of empty books.

About the linter: it is sending false errors cause its detecting my functions as unused. But they are directly called from my HTML.

An easy way to fix this is to just call them at the end of the script but that's unnecessary and actually a bad practice on a real world scenario.  

Also no nested ternary operators make no sense. You can read why nested ternary operators are good practice here:  https://hackernoon.com/the-untapped-potential-of-nested-ternaries-2cb98079b634 and here: https://medium.com/javascript-scene/nested-ternaries-are-great-361bddd0f340

I find them more readable and modern than if-else.

Maybe as a reviewer you can report of this issues so changes are made to the config file?

For that if you compare my config with the original I just removed the unused-var rule and the ternary rule. 


Cheers!

**Review by shloch — CHANGES_REQUESTED** [review]
## hello @Luzaks , @MarioDena 

- Please the remark of previews review is still unchanged, "empty" books can be added on your app, this shouldn't be possible
Screen Shot 2020-03-21 at 8 46 57 AM

- Also, you might want to apply some controls to ensure that `pages` is a number
- Finally, when i click on adding NEW FORM, see the form that appears (2 in 1). This doesn't seem intuitively normal

Screen Shot 2020-03-21 at 8 45 59 AM

### please, debug these

**Review by shloch — CHANGES_REQUESTED** [review]
## hello @Luzaks , @MarioDena 

Great job correcting the form that appears

Screen Shot 2020-03-21 at 11 13 31 AM

### [required changes]

- when user validates empty form..look for a way to show an error message that form is empty (make this user friendly)
- app doesn't work again, i can't add any book (please debug this)
- dont forget to verify that `pages` is a number

**Comment by MarioDena** [comment]
## I can add books but still changed the required code.... double check maybe?

I can't find where is it specified that we need to show an error message or verify that the pages only take numeric input. (the input type is number)

Your requested changes are not found Here: https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623451/

or Here: https://www.theodinproject.com/courses/javascript/lessons/library

Or here: https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/
 

Please double check your mandatory requirements.

File: script.js [hunk 2]
```diff
@@ -86,7 +86,7 @@ function addBookToLibrary() {
 }
 
 function checkCreatedValid() {
-  if (getId('title').value === true) {
+  if (getId('title').value !== '') {
     addBookToLibrary();
   }
 }
```
Comment (shloch) [inline script.js]: It will be good to validate not only title, but maybe the other fields too

**Review by shloch — APPROVED** [review]
## hello @Luzaks , @MarioDena 

Great job....all requirements met.

I can add books now

### it is not obligatory but don't you think it's reasonable like all other apps to let users know when they validate an empty form instead of noticing nothing on screen except form disappears??. No It's part of design.

Think about it. I would give you a thumbs up, but that would be a great to have  😊😊

File: script.js (hunk 2 above)
Comment (shloch) [reply script.js]: We're here to help ourselves makes apps as professional as possible
Screen Shot 2020-03-21 at 12 03 35 PM

**Comment by MarioDena** [comment]
@shloch Yes of course is reasonable but not required on the project.

The thing is you are overlooking other extra features we added to the project. For example more status other than read and not read. DRY code using a function to avoid repeating DOM element finding, custom pop ups for editing and creating, opacity on pop up show... Custom styling (also not mandatory)...

Of course we noticed the blank books, and numeric input validations. but we went for more interesting extra features (At least for us).

That's why it makes me feel frustrated to see non-mandatory requirements getting our project rejected. 

I will implement your changes anyway. Thank you for your input and  feedback.

**Comment by shloch** [comment]
i should have flagged it as **optional**, your right 😊😊
I wont forget next time...happy coding

**Pull request merged**